### PR TITLE
Fix several problems with using rkt on gce

### PR DESCRIPTION
**Pull request by runningwild**
With these changes in I am able to bring up a rkt cluster on gce and all nodes appear healthy.  I have yet to actually run a rkt container on the cluster, though, just going one step at a time.

At the very least this addresses issue #11824, and unlike PR #12808, I believe this should pass all tests (fingers crossed).

### CODE REVIEW

**Comment by vishh** [comment]
Thanks for the PR @runningwild! Overall LGTM. Can you fix the test failures? It might be worth adding an e2e test for CoreOS and rocket support. WDYT?

**Comment by runningwild** [comment]
Ok looks like I can just push a new branch and this will pick it up, that's pretty cool.

I'll work on getting that test fixed.  Until I'm sure what I have to run to test locally pushing is the best way to know if everything is working.

An e2e test sounds great, not sure how to write those though.  What would be the best place to look for inspiration?

Also, dchen1107 wants me to assign this to her but I'm not sure how, I don't think I have that power.

**Comment by yifan-gu** [comment]
@runningwild Thanks for the PR! FWIW the new flag LGTM. I am just about to do this in the next couple of PRs (https://github.com/kubernetes/kubernetes/pull/8951#issuecomment-130843046)

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -245,6 +247,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.ResourceContainer, "resource-container", s.ResourceContainer, "Absolute name of the resource-only container to create and run the Kubelet in (Default: /kubelet).")
 	fs.StringVar(&s.CgroupRoot, "cgroup-root", s.CgroupRoot, "Optional root cgroup to use for pods. This is handled by the container runtime on a best effort basis. Default: '', which means use the container runtime default.")
 	fs.StringVar(&s.ContainerRuntime, "container-runtime", s.ContainerRuntime, "The container runtime to use. Possible values: 'docker', 'rkt'. Default: 'docker'.")
+	fs.StringVar(&s.RktPath, "rkt-path", s.RktPath, "Path of rkt binary. Leave empty to use the first rkt in $PATH")
```
Comment (dchen1107) [inline cmd/kubelet/app/server.go]: nit: in flag description, indicates that the flag is ignored if --container-runtime is not set to rkt.

**Comment by dchen1107** [comment]
LGTM except above nit

**Comment by runningwild** [comment]
Changed flag comment.  I used a positive clause ("Only used if --container-runtime=='rkt'") rather than a negative one ("Not used if --container-runtime!='rkt'"), there are both kinds already in the flags descriptions so I figured this was ok.

**Pull request merged**